core_types/src/agent.rs

```diff
@@ -7,53 +7,56 @@ use crate::{
     json::JsonString,
 };
 
-use std::convert::TryFrom;
+use std::{convert::TryFrom, str};
 
 use crate::error::HolochainError;
 use reed_solomon::{Decoder, Encoder};
 
-const PARITY_LEN: usize = 2;
-
 /// A raw public key buffer
 /// Can extract the signature and encryption portions
 /// Can parse a base64url encoded user representation
 /// Can render a base64url encoded user representation
 #[derive(Clone)]
-pub struct KeyBuffer([u8; 64]);
+pub struct KeyBuffer([u8; KeyBuffer::KEY_LEN]);
 
 impl KeyBuffer {
+    /// Constants specific to KeyBuffer
+    const PARITY_LEN: usize = 5;
+    const KEY_LEN: usize = 64;
+    const HALF_KEY_LEN: usize = KeyBuffer::KEY_LEN / 2;
+
     /// take a potentially user-entered base64url encoded user representation
     /// of an public key identity
     /// apply reed-solomon parity correction
     /// returns a raw byte buffer
     pub fn with_corrected(s: &str) -> Result<KeyBuffer, HolochainError> {
         let s = s.replace("-", "+").replace("_", "/");
-        let s = base64::decode(&s)?;
-        let dec = Decoder::new(PARITY_LEN);
-        let dec = *dec.correct(s.as_slice(), None)?;
-        Ok(KeyBuffer::with_raw(array_ref![dec, 0, 64]))
+        let base64 = base64::decode(&s)?;
+        let dec = Decoder::new(KeyBuffer::PARITY_LEN);
+        let dec = *dec.correct(base64.as_slice(), None)?;
+        Ok(KeyBuffer::with_raw(array_ref![dec, 0, KeyBuffer::KEY_LEN]))
```

Comment:
Reed-Solomon will erroneously correct things if it uses up all its parity confidence. @pjkundert can give further details, but I think we need to add something like

```rust
const CORRECTION_BYTE_LIMIT = 2
```

and return a HolochainError if > this number of bytes are altered compared to the input string